cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/validator/ApacheValidator.java

```diff
@@ -0,0 +1,389 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.template.etl.validator;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.template.etl.api.Validator;
+import org.apache.commons.validator.GenericValidator;
+import org.apache.commons.validator.routines.CreditCardValidator;
+import org.apache.commons.validator.routines.DateValidator;
+import org.apache.commons.validator.routines.DomainValidator;
+import org.apache.commons.validator.routines.DoubleValidator;
+import org.apache.commons.validator.routines.EmailValidator;
+import org.apache.commons.validator.routines.FloatValidator;
+import org.apache.commons.validator.routines.ISBNValidator;
+import org.apache.commons.validator.routines.InetAddressValidator;
+import org.apache.commons.validator.routines.IntegerValidator;
+import org.apache.commons.validator.routines.LongValidator;
+import org.apache.commons.validator.routines.RegexValidator;
+import org.apache.commons.validator.routines.ShortValidator;
+import org.apache.commons.validator.routines.UrlValidator;
+
+/**
+ * Utility class containing validator functions provided by org.apache.commons.validator.routines package.
+ */
+@Plugin(type = "validator")
+@Name("apache")
+@Description("Apache validator functions from org.apache.commons.validator.routines package. " +
+  "Includes isDate(String), isCreditCard(String), isBlankOrNull(String), isEmail(String), " +
+  "isInRange (value, min, max)[supported types : double, int, float, short]," +
+  "isInt(String), isLong(String), isShort(String), isUrl(String), matchRegex(String pattern, String input)," +
+  "maxLength(String input, int maxLength)," +
+  "maxValue(val, maxValue)[supported types : double, int, float, short]," +
+  "minLength(String input, int maxLength)," +
+  "minValue(val, minValue)[supported types : double, int, float, short]," +
+  "isValidISBN(String), isValidInet4Address(String), isValidInet6Address(String), isValidIp(String)" +
+  "isValidCountryCodeTid(String), isValidGenericTId(String), " +
+  "isValidInfrastructureTId(String),isValidLocalTId(String)," +
+  "isValidTId(String)")
+public class ApacheValidator implements Validator {
+
+  @Override
+  public String getValidatorVariableName() {
+    return "apacheValidator";
+  }
+
+  @Override
+  public Object getValidator() {
+    return new ApacheValidatorWrapper();
+  }
+
+  /**
+   * Wrapper class for apache validator functions.
+   */
+  public static final class ApacheValidatorWrapper {
+
+    /**
+     * return true if the passed param is a valid date.
+     * uses {@link org.apache.commons.validator.routines.DateValidator} to check validity.
+     * @param date
+     * @return boolean
+     */
+    public static boolean isDate(String date) {
```

Comment:
Are these methods supposed to be called on the instance returned by ``getValidator``? If so, why are they static?